app/src/main/java/com/duckduckgo/app/fire/DataClearerForegroundAppRestartPixel.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.fire
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.edit
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import com.duckduckgo.app.global.intentText
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.systemsearch.SystemSearchActivity
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Stores information about unsent automatic data clearer restart Pixels, detecting if user started the app from an external Intent.
+ * Contains logic to sent unsent pixels.
+ *
+ * When writing values here to SharedPreferences, it is crucial to use `commit = true`. As otherwise the change can be lost in the process restart.
+ */
+@Singleton
+class DataClearerForegroundAppRestartPixel @Inject constructor(
+    private val context: Context,
+    private val pixel: Pixel
+) : LifecycleObserver {
+    private var detectedUserIntent: Boolean = false
+
+    private val pendingAppForegroundRestart: Int
+        get() = preferences.getInt(KEY_UNSENT_CLEAR_APP_RESTARTED_PIXELS, 0)
+
+    private val pendingAppForegroundRestartWithIntent: Int
+        get() = preferences.getInt(KEY_UNSENT_CLEAR_APP_RESTARTED_WITH_INTENT_PIXELS, 0)
+
+    @UiThread
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onAppBackgrounded() {
+        Timber.i("Registered App on_stop")
+        detectedUserIntent = false
+    }
+
+    fun registerIntent(intent: Intent?) {
```

Comment:
can we find a better name for this? it's not clear from the name what it does

Reply:
I could come up with a better name, like I previously expected. No problem since this will be removed once we work on the fix at some point.